tests/cross/src/main/scala/tests/BasePCSuite.scala

```diff
@@ -67,25 +71,31 @@ abstract class BasePCSuite extends BaseSuite {
     JdkSources().foreach(jdk => index.addSourceJar(jdk))
   }
 
-  override def test(options: TestOptions)(fun: => Any): Unit = {
-    // We are unable to infer the JDK jars on Appveyor
-    // tests.BasePCSuite.indexJDK(BasePCSuite.scala:44)
+  override def munitNewTest(test: Test): Test = {
     val testName =
       if (isCI && BuildInfo.scalaCompilerVersion != BuildInfoVersions.scala212)
-        s"${BuildInfo.scalaCompilerVersion}-${options.name}"
-      else options.name
-    super.test(options.copy(name = testName)) {
-      try {
-        fun
-      } catch {
-        case NonFatal(e)
-            if e.getMessage != null &&
-              e.getMessage.contains("x$1") &&
-              !hasJdkSources =>
-          // ignore failing test if jdk sources are missing
-          ()
+        s"${BuildInfo.scalaCompilerVersion}-${test.name}"
+      else test.name
+    test.withName(testName)
+  }
+
+  override def munitRunTest(options: TestOptions, body: => Any): Any = {
```

Comment:
I think this is a good chance to try removing this: it may not be needed on GitHub actions